src/main/frontend/hooks/useSamples.ts

```diff
@@ -0,0 +1,39 @@
+import { Sample, SAMPLES } from '../model/samples';
+import { useEffect, useState } from 'react';
+import { Script } from '../model/Script';
+
+export const useSamples = (): [Sample[], boolean] => {
+	const [samples, setSamples] = useState<Sample[]>([]);
+	const [loading, setLoading] = useState(false);
+
+	useEffect(() => {
+		const loadSamples = async () => {
+			setLoading(true);
+			try {
+				const response = await fetch('/apps/jcr-hopper/script-builder/samples/additional-samples.json');
```

Comment:
I would prefer the server to inject the samples into the HTML (e.g. as a data attribute).

Reply:
I think this makes sense, especially if we move all of the scripts/samples to a folder as mentioned below.